packages/embark/src/lib/core/plugin.js

```diff
@@ -229,14 +230,17 @@ Plugin.prototype.registerUploadCommand = function(cmd, cb) {
 
 Plugin.prototype.addCodeToEmbarkJS = function(code) {
   this.addPluginType('embarkjsCode');
+  // TODO: what is this/why
   if (!this.embarkjs_code.some((existingCode) => deepEqual(existingCode, code))) {
     this.embarkjs_code.push(code);
-    this.events.request('blockchain:ready', () => {
-      this.events.emit('runcode:embarkjs-code:updated', code, () => {});
-    });
   }
 };
 
+Plugin.prototype.addGeneratedCode = function(codeCb) {
+  this.addPluginType('generatedCode');
+  this.generated_code.push(codeCb);
+};
```

Comment:
Is this supposed to be a public API?

If yes, this needs documentation.

Reply:
In any case it's a new API and @emizzle's `embark[js]-omg` packages will need to be refactored to use it. @andremedeiros and I discussed briefly whether it constitutes a major-version breaking change. We tentatively concluded that the code generation APIs/events are an internal implementation and so introducing a change is unlikely to affect the vast majority of our users, with the exception of experts and team members who were hooking into our internals. What do you think @PascalPrecht?

Reply:
I wasn't aware that it would be a breaking change, but it also doesn't seem to be a plugin API exposed for users that build plugins, right? In that case, I would consider this thing not to be a public API.